openslides/utils/autoupdate.py

```diff
@@ -92,6 +94,12 @@ def ws_add_site(message):
         access_permissions = collection.get_access_permissions()
         restricted_data = access_permissions.get_restricted_data(collection, user)
 
+        if restricted_data is None:
```

Comment:
This can should never happen: In line 93 (old) / 95 (new) we get the restricted data using a Collection And not a single instance. In this case all restricted_data functions have to return an iterable (e. g. an empty list).

Is it possible to remove the two lines?

Reply:
Yes, it is possible. The cast() call in the next line should solve this. But please be aware of this issue: https://github.com/OpenSlides/OpenSlides/issues/3377

My goal is still to remove the "container" object in the get_restricted_data methods and replace it with List[Dict[str, Any]] (maybe the value in the dict could also be str, or Unique[str, int, bool])

Reply:
Issue #3377 is fixed in current master, so please remove these lines.

Reply:
done